src/gravel/controllers/nodes/systemdisk.py

```diff
@@ -0,0 +1,255 @@
+
+        # ascertain whether we can use this device
+        if not device:
+            raise UnknownDeviceError(f"device {devicestr} not known")
+        elif not device.available:
+            raise UnavailableDeviceError(f"device {devicestr} is not available")
+
+        # create partitions
+        logger.debug(f"prepare system disk: device: {device}")
+        devpath = device.path
+
+        def _create_overlay_dir(path: Path, dirname: str) -> None:
+            assert path.exists()
+            dirpath: Path = path.joinpath(dirname)
+            assert not dirpath.exists()
+            dirpath.mkdir()
+            dirpath.joinpath("overlay").mkdir()
+            dirpath.joinpath("temp").mkdir()
+
+        try:
+            # create lvm volume
+            await self.lvm(f"pvcreate {devpath}")
+            await self.lvm(f"vgcreate aquarium {devpath} --addtag @aquarium")
+            await self.lvm("lvcreate -l 100%FREE -n systemdisk aquarium")
+
+            lvmdev: Path = Path("/dev/mapper/aquarium-systemdisk")
+            assert lvmdev.exists()
+
+            # format with xfs
+            await aqr_run_cmd(shlex.split(f"mkfs.xfs -m bigtime=1 {lvmdev}"))
+
+            aqrmntpath: Path = Path("/aquarium")
+            aqrmntpath.mkdir(exist_ok=True)
+
+            await self.mount()
+
+            for d in self._overlaydirs.keys():
+                _create_overlay_dir(aqrmntpath, d)
+
+            # some directories (like ceph's) can't be overlayed due to weirdness
+            # ensued, so we'll bind mount them later.
+            for d in self._bindmounts.keys():
+                ourpath: Path = aqrmntpath.joinpath(d)
+                assert not ourpath.exists()
+                ourpath.mkdir()
+
+            await self.unmount()
+
+        except Exception as e:
+            logger.error(f"prepare system disk > {str(e)}")
+            logger.exception(e)
+            raise e
+
+    async def mount(self) -> None:
+        aqrmntpath: Path = Path("/aquarium")
+        aqrdevpath: Path = Path("/dev/mapper/aquarium-systemdisk")
+        assert aqrdevpath.exists()
+        aqrmntpath.mkdir(exist_ok=True)
+
+        aqrdev: str = aqrdevpath.as_posix()
+        aqrmnt: str = aqrmntpath.as_posix()
+
+        try:
+            await aqr_run_cmd(shlex.split(f"mount {aqrdev} {aqrmnt}"))
+        except Exception as e:
+            raise MountError(msg=str(e))
+
+    async def unmount(self) -> None:
+        aqrmnt: Path = Path("/aquarium")
+        if not aqrmnt.exists():
+            raise MountError(msg="/aquarium mount point does not exist")
+
+        try:
+            await aqr_run_cmd(shlex.split(f"umount {aqrmnt}"))
+        except Exception as e:
+            raise MountError(msg=str(e))
+
+    # lacking a better name atm, we're calling this function 'swap in', as in
+    # we're swapping these directories in, some being overlayed, others bind
+    # mounted.
```

Comment:
Do we need a `swapout` as well?

(nit) An alternative could be `enable`?

Reply:
yeah, sounds like a good alternative.